docs(docs): fixed duplicate IDs in example docs

evwilkin opened the pull request:
Closes #4927 

This PR resolves duplicate IDs throwing a11y errors identified in https://github.com/patternfly/patternfly-org/issues/2920

patternfly-build commented:
Preview: https://patternfly-pr-4928.surge.sh

nicolethoen commented on src/patternfly/components/Pagination/examples/Pagination.md:
```diff
@@ -29,7 +29,7 @@ import './Pagination.css'
 {{#> pagination pagination--modifier="pf-m-sticky"}}
   {{> pagination-total-items-content}}
-  {{> pagination-options-menu id="pagination-options-menu-top-example"}}
+  {{> pagination-options-menu id="pagination-options-menu-top-sticky-example"}}
```
While we are in here, most of the example are all using the same aria-label on the `.pf-c-pagination__nav` of "Pagination". Could these all be updated to be unique to the example?

mcoker commented on src/patternfly/components/Select/examples/Select.md:
````diff
@@ -445,14 +445,14 @@ The plain select variation should be used when you do not want a border applied
 ### Placeholder item disabled
 ```hbs
-{{#> select id="select-placeholder-expanded" select--IsExpanded="true" select--IsPlaceholder="true" select--HasPlaceholderItem="true" select--PlaceholderItemDisabled="true"}}
+{{#> select id="select-placeholder-item-disabled" select--IsExpanded="true" select--IsPlaceholder="true" select--HasPlaceholderItem="true" select--PlaceholderItemDisabled="true"}}
````
This is out of scope, but it would be nice to replace `id` with `select--id` in the hbs for this component. If you'd like to do that, feel free to create a separate issue. It could be a lot to update though since we have selects in a lot of our examples and demos.

evwilkin replied:
Created https://github.com/patternfly/patternfly/issues/4999 as a follow-up to address this.

mcoker commented on src/patternfly/components/TextInputGroup/examples/TextInputGroup.md:
````diff
@@ -83,7 +83,7 @@ import './TextInputGroup.css'
 
 ### Autocomplete last option hint
 ```hbs
-{{#> text-input-group text-input-group--id="basic" text-input-group--value="apples"}}
````
I see you removed `text-input-group--id` here. It looks like the only place we reference `text-input-group--id` is when there is a chip group - https://github.com/patternfly/patternfly/blob/e530391ee29e1b9f719b40881b635914a99fb80e/src/patternfly/components/TextInputGroup/templates/text-input-group--chip-group.hbs#L1

So while it's not needed here, we have `text-input-group--id` in other examples that do not have a chip group. I think we should be consistent - I'd be in favor of adding `text-input-group--id` to all of the examples whether they have a chip group or not, just so it's there if we want to use it.

mcoker commented on src/patternfly/components/TextInputGroup/templates/text-input-group--chip-group.hbs:
```diff
@@ -1,143 +1,143 @@
-{{#> chip-group chip-group--id=(concat text-input-group--id '-chip-group')}}
```
I'm assuming the logic having the `--id` set at the level of the chip group is in case this file includes multiple chip groups - each would be able to pass a unique `--id` to its chips. Using `text-input-group--id` on the chips would end up with dupes if there was more than one chip group.

mcoker commented on src/patternfly/components/TextInputGroup/text-input-group.hbs:
```diff
@@ -5,6 +5,9 @@
     {{text-input-group--modifier}}
   {{/if}}"
+  {{#if text-input-group--id}}
+    id="{{text-input-group--id}}"
```
do you need this?

mcoker commented on src/patternfly/demos/Toolbar/examples/Toolbar.md:
```diff
@@ -52,7 +52,7 @@ import './Toolbar.css'
       {{#> toolbar-item toolbar-item--modifier=(concat 'pf-m-pagination ' toolbar-item-pagination--modifier)}}
         {{#> pagination pagination--modifier="pf-m-compact pf-m-hidden pf-m-visible-on-md"}}
           {{#> pagination pagination--IsCompact="true"}}
-            {{> pagination-options-menu id="pagination-options-menu-bottom-example"}}
```
Same comment as mentioned about the select component for updating `id` to `options-menu--id`. Sharing the same partial parameter name can cause issues.

evwilkin replied:
Created https://github.com/patternfly/patternfly/issues/5000 as follow-up to track this request.

mcoker commented on src/patternfly/demos/Toolbar/examples/Toolbar.md:
```diff
@@ -187,30 +187,30 @@ import './Toolbar.css'
             {{#> chip-group-list-item}}
               {{#> chip}}
-                {{#> chip-text chip-text--attribute=(concat 'id="' chip-group--id 'chip-one"')}}
+                {{#> chip-text chip-text--attribute=(concat 'id="' toolbar--id 'chip-one"')}}
```
This works since there is only a single chip group in this toolbar, but we have a couple of examples with a toolbar that has multiple chip groups - https://www.patternfly.org/v4/components/toolbar/html#selected-filters-on-mobile-filters-collapsed-expandable-content-expanded

IMO it would save us some work later to leave these as referencing `chip-group--id`, and also update the `{{#> chp-group-label` and `{{#> chip-group-list` params to reference `chip-group--id`, in case we add more chip groups to these toolbar demos.

mcoker reviewed (changes requested):
So much better!! Left some comments, let me know what you think!

evwilkin commented on src/patternfly/components/MultipleFileUpload/multiple-file-upload.hbs:
```diff
@@ -1,4 +1,7 @@
 <div class="pf-c-multiple-file-upload{{#if multiple-file-upload--modifier}} {{multiple-file-upload--modifier}}{{/if}}"
+  {{#if multiple-file-upload--id}}
+    id="{{multiple-file-upload--id}}"
```
@mcoker thanks for calling out - I thought a variable passed through the `.md` file needed to be referenced directly in order to be used by nested handlebar templates but see that's not the case, so no these can be removed as an unneeded "intermediate" steps between the .md file & the nested .hbs file that uses the variable.

mcoker commented on src/patternfly/components/AppLauncher/app-launcher.hbs:
```diff
@@ -1,5 +1,8 @@
 <nav class="pf-c-app-launcher{{#if app-launcher--IsExpanded}} pf-m-expanded{{/if}}{{#if app-launcher--modifier}} {{app-launcher--modifier}}{{/if}}"
   aria-label="Application launcher"
+  {{#if app-launcher--id}}
+    id="{{app-launcher--id}}"
+  {{/if}}
```
Do you need this?

```suggestion
```

evwilkin replied:
Nope - thanks for catching this!

mcoker commented on src/patternfly/components/DataList/examples/DataList.md:
````diff
@@ -296,7 +296,7 @@ When a list item includes more than one block of content, it can be difficult fo
 ### Expandable compact
 ```hbs
-{{#> data-list data-list--modifier="pf-m-compact" data-list--id="data-list-expandable" data-list--attribute='aria-label="Expandable data list example"'}}
+{{#> data-list data-list--modifier="pf-m-compact" data-list--id="data-list-expandable-compact" data-list--attribute='aria-label="Expandable data list example"'}}
````
"[secondary/tertiary] content details" both show up 4 times in the expandable examples, too. Is this an axe error message? I don't see it when I run it in my browser. I imagine this would be a pretty widespread issue.

nicolethoen replied:
I think it would be flagged as an issue - but the aria-labels in the expandable examples are being generated incorrectly. For example, the first expandable content in the expandable compact example has an aria-label= `data-list-expandable-compact Primary content details`

mcoker commented on src/patternfly/components/Tabs/examples/Tabs.md:
````diff
@@ -199,18 +199,18 @@ import './Tabs.css'
 ### Using the nav element
 ```hbs
-{{#> tabs tabs--id="default-scroll-nav-example" tabs--type="nav" tabs--modifier="pf-m-scrollable" tabs--attribute='aria-label="Local"' tabs-link--isLink="true"}}
+{{#> tabs tabs--id="default-scroll-nav-example" tabs--type="nav" tabs--modifier="pf-m-scrollable" tabs--attribute='aria-label="Tabs nav"' tabs-link--isLink="true"}}
````
We have this same duplication of the `aria-label` in the navigation component examples, too, and you can see that "Local" or "Global" is auto-generated as the `aria-label` in the `<Nav>` component depending on the type of nav used - https://github.com/patternfly/patternfly-react/blob/7bfaee093d3c85392a9d21a34ff661ffe29b0f47/packages/react-core/src/components/Nav/Nav.tsx#L162

I'm curious if that's an issue as well?

nicolethoen replied:
On the react side, I overrode the default aria-label in the examples to avoid the duplicate aria label issue. I figure in product, the likelihood that they will run into duplicate labels is unlikely, and many products have to localize and override aria labels anyway.

evwilkin replied:
@nicolethoen can you clarify - I do see these duplicate aria label errors throw for "Global" and "Local", should we change here to fix those errors?  And if so do you have a recommendation, as it sounds like these specific labels have additional meaning

mcoker commented on src/patternfly/demos/Toolbar/examples/Toolbar.md:
```diff
@@ -179,15 +179,15 @@ import './Toolbar.css'
   {{#> toolbar-content toolbar-content--modifier="pf-m-chip-container"}}
     {{#> toolbar-item toolbar-item--modifier=(concat 'pf-m-chip-group ' toolbar-item-chip-group--modifier)}}
-      {{#> chip-group chip-group--modifier="pf-m-category"}}
+      {{#> chip-group chip-group--modifier="pf-m-category" chip-group--id="toolbar-attribute-value-checkbox-select-filter-desktop-example-chip-group"}}
```
could you do this?

```suggestion
      {{#> chip-group chip-group--modifier="pf-m-category" chip-group--id=(concat toolbar--id "-chip-group")}}
```

evwilkin replied:
That's much more graceful 👍

mcoker commented on src/patternfly/demos/Toolbar/examples/Toolbar.md:
```diff
@@ -255,12 +255,12 @@ import './Toolbar.css'
         {{/toolbar-item}}
         {{#> toolbar-item toolbar-item--modifier=(concat 'pf-m-chip-group ' toolbar-item-chip-group--modifier)}}
-          {{#> chip-group chip-group--modifier="pf-m-category"}}
+          {{#> chip-group chip-group--modifier="pf-m-category" chip-group--id="toolbar-attribute-value-checkbox-select-filter-mobile-example-chip-group"}}
```
```suggestion
          {{#> chip-group chip-group--modifier="pf-m-category" chip-group--id=(concat toolbar--id "-chip-group")}}
```

mcoker reviewed (commented):
Looks great - left a few comments!

evwilkin commented on src/patternfly/components/Nav/examples/Navigation.md:
```diff
@@ -685,10 +685,10 @@ import './Navigation.css'
     {{/nav-item}}
     {{#> nav-item nav-item--expandable="true" nav-item--expanded="true"}}
-      {{#> nav-link nav-link--href="#" nav-link--attribute='id="expandable-example2"'}}
+      {{#> nav-link nav-link--href="#" nav-link--attribute='id="expandable-light-example2"'}}
```
Created https://github.com/patternfly/patternfly/issues/5008 to track this separately due to the size of this effort and difficulty naming the many different examples on the Nav page

evwilkin commented:
@nicolethoen @mcoker I believe I've addressed all feedback here other than [this question on Tabs](https://github.com/patternfly/patternfly/pull/4928#discussion_r932589207) and the comment regarding Nav example text that I split out into a separate issue.  Thanks for following along on this adventure!

mcoker reviewed (approved):
🤩

patternfly-build commented:
:tada: This PR is included in version 4.206.0 :tada:

The release is available on:
- [GitHub release](https://github.com/patternfly/patternfly/releases/tag/prerelease-v4.206.0)
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@patternfly/patternfly/v/4.206.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
